concepts/union-type/about.md

```diff
@@ -0,0 +1,152 @@
+# Union types
+
+Crystal allows for a variable to consist of multiple types.
+This is called a [union type][union-type].
+In Crystal is it quite often a union type inferred by the compiler.
+
+```exercism/note
+A union type even if it consists of multiple types is still a single type at runtime.
+Meaning a union type is built of String and Int32 so will it not be both at the same time.
+Instead it will be either a String or an Int32.
+```
+
+A union type is declared by separating the types with a pipe (`|`).
+They are often placed in parenthesis, but it is not required.
+The most common union type is `(T | Nil)` where `T` is a type, which can occur when doing methods that can return `Nil`.
+This behavior is also known as a variable being **nilable**.
+
+```crystal
+a : String = "Hello"
+a = nil # Error: type must be String, not (String | Nil)
+
+a : (String | Nil) = "Hello"
+a = nil
+```
+
+It is not limited to just 2 types, but can be as many as you want.
+
+```crystal
+a : (String | Int32 | Nil | Float64) = "Hello"
+a = 1
+a = nil
+a = 1.0
+```
+
+## `typeof` vs `Object#class`
+
+There are 2 ways to get the type of a variable.
+Either by using [`typeof`][typeof] or by using [`Object#class`][Object#class].
+The difference between them is that typeof will return a variables type at compile time, while `Object#class` will return the type at runtime.
+Meaning if you want to for example see if a variable is a union type, then will `Object#class` not be able to tell you that, as it will only return the type at runtime, which is a single type.
+
+```crystal
+a : (String | Int32) = "Hello"
+typeof(a) # => (String | Int32)
+a.class # => String
+```
+
+## Operations on union types
+
+As a union type is a single type at runtime, so does all the normal operations work on it.
+But when compiling the code will the compiler not know which type it is.
+Thereby the code has to be setup in such a way that it can only be one of the types, when wanting to use the type specific operations.
```

Comment:
```suggestion
Thereby the code has to be setup in such a way that it can only be one of the types when wanting to use the type specific operations.
```